### ISSUE TITLE
Missing package.json mappings for itwinui-react/react-table while using "moduleResolution": "node16"

### ISSUE THREAD

**Issue by Antanas-Bentley** [iTwin/iTwinUI#2072]
### Describe the bug (current behavior)

After updating to v3, the package we are consuming iTwinUI in cannot resolve paths to import anything from '@itwin/itwinui-react/react-table':
![image](https://github.com/iTwin/iTwinUI/assets/159042330/a4aa4b3a-08c5-4dc0-a1c8-fc95e1cbe830)

Looks like we cannot resolve the import when the current file is detected to use CJS based on tsconfig. We are using `"moduleResolution": "node16"` which appears to be causing the issue as this `moduleResolution` cannot find `itwinui-react/react-table` paths.

### Expected Behavior

We should be able to use 'import' syntax for '@itwin/itwinui-react/react-table' elements.

### Link to minimal repro

https://stackblitz.com/edit/github-lw2hd6?file=src%2FApp.tsx

### Steps To Reproduce

1. Copy the tsconfig provided in the minimal repo (with `"moduleResolution": "node16"`)
2. Try to import anything from '@itwin/itwinui-react/react-table'

**Comment by Antanas-Bentley** [comment]
Adding explicit import paths fixes the issue:
```
    "./react-table": {
      "import": {
        "types": "./esm/react-table/react-table.d.ts",
        "default": "./esm/react-table/react-table.js"
      },
      "require": {
        "types": "./cjs/react-table/react-table.d.ts",
        "default": "./cjs/react-table/react-table.js"
      }
    },
```

**Comment by mayank99** [comment]
Fixed in [3.11.1](https://github.com/iTwin/iTwinUI/releases/tag/%40itwin%2Fitwinui-react%403.11.1) 🥳 Thanks for submitting this issue and the PR @Antanas-Bentley
